dadi/lib/models/route.js

```diff
@@ -265,18 +267,19 @@ Route.prototype.processRoute = function () {
   })
 }
 
-Route.prototype.save = function () {
-  const filePath = path.join(path.resolve(config.get('paths.routes')), this.config.route + '.json')
-
-  try {
-    fs.writeFileSync(filePath, JSON.stringify(this.config, null, 2))
-
-    return true
-  } catch (err) {
-    logger.error({module: 'routes'}, err)
-
-    return false
-  }
+Route.prototype.save = function (domainName) {
+  let domain = domainManager.getDomain(domainName)
+  let routePath = path.resolve(
+    path.join(
+      domain ? domain.path : '',
+      config.get('paths.routes', domainName),
+      `${this.config.route}.json`
+    )
+  )
+
+  return fs.writeJson(routePath, this.config, {
```

Comment:
what's the benefit of using fs-extra.writeJson over fs.writeFile here?

Reply:
The benefit of `fs-extra` is that it augments `fs` with support for Promises (whilst retaining backward-compatibility), which makes the code much shorter and easier to read, in my opinion (as opposed to us wrapping every `fs` method in a `Promise`). Using `writeJson` over `writeFile` is just a convenience method, really, as it takes care of stringifying the payload and handling errors.